Sources/App/WebView/WebViewWindowController.swift

```diff
@@ -202,9 +202,31 @@ class WebViewWindowController {
         }
     }
 
-    func openSelectingServer(from: OpenSource, urlString openUrlRaw: String, skipConfirm: Bool = false) {
-        if let first = Current.servers.all.first, Current.servers.all.count == 1 {
-            open(from: from, server: first, urlString: openUrlRaw, skipConfirm: skipConfirm)
+    func openSelectingServer(
+        from: OpenSource,
+        urlString openUrlRaw: String,
+        skipConfirm: Bool = false,
+        queryParameters: [URLQueryItem]? = nil
+    ) {
+        let serverName = queryParameters?.first(where: { $0.name == "server" })?.value
+        let servers = Current.servers.all
+
+        if let first = servers.first, Current.servers.all.count == 1 || serverName != nil {
+            if serverName == "default" || serverName == nil {
+                open(from: from, server: first, urlString: openUrlRaw, skipConfirm: skipConfirm)
+            } else {
+                if let selectedServer = servers.first(where: { server in
+                    let sanitizedServerName = server.info.name.lowercased()
+                        .addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed)
+                    let sanitizedSelectedServerName = serverName?.lowercased()
+                        .addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed)
+                    return sanitizedServerName == sanitizedSelectedServerName
```

Comment:
can we fuzzy match? for example, if i have:

1. `zac's house`
2. `the shed`

can we match with `?server=house` and `?server=shed`?

Reply:
It's doable but should we? I think this may lead to name conflict at some point

Reply:
Like If I have 2 server "House" and "Parent's House", using fuzzy match on "House" will conflict

Reply:
hm, i think something like `server=house` would match `house` over `parent's house` -- maybe the 'closer' the better. maybe that's good for a second round at this.